resources/mediaprocessor.py

```diff
@@ -1490,64 +1494,85 @@ def checkDisposition(self, allowed, source):
-                if decoder in codecs[video_codec]['decoders'] and decoder in self.settings.hwaccel_decoders:
-                    if Converter.decoder(decoder).supportsBitDepth(bit_depth):
-                        self.log.info("%s decoder is also supported by this ffmpeg build and will also be used [hwaccel-decoders]." % decoder)
-                        opts.extend(['-vcodec', decoder])
-                        self.log.debug("Decoder formats:")
-                        self.log.debug(self.converter.ffmpeg.decoder_formats(decoder))
-                    else:
-                        self.log.debug("Decoder %s is supported but cannot support bit depth %d of format %s." % (decoder, bit_depth, pix_fmt))
-                break
-        if "-vcodec" not in opts:
-            # No matching decoder found for hwaccel, see if there's still a valid decoder that may not match
-            for decoder in self.settings.hwaccel_decoders:
-                if decoder in codecs[video_codec]['decoders'] and decoder in self.settings.hwaccel_decoders and decoder.startswith(video_codec):
-                    if Converter.decoder(decoder).supportsBitDepth(bit_depth):
-                        self.log.info("%s decoder is supported by this ffmpeg build and will also be used [hwaccel-decoders]." % decoder)
-                        opts.extend(['-vcodec', decoder])
-                        self.log.debug("Decoder formats:")
-                        self.log.debug(self.converter.ffmpeg.decoder_formats(decoder))
-                        break
-                    else:
-                        self.log.debug("Decoder %s is supported but cannot support bit depth %d of format %s." % (decoder, bit_depth, pix_fmt))
+
+        self.log.debug(f"Selected hwaccel options: {self.settings.hwaccels}.")
+        self.log.debug(f"Selected hwaccel decoder pairs: {self.settings.hwaccel_decoders}.")
+        self.log.debug(f"Supported ffmpeg hwaccels: {hwaccels}.")
+        self.log.debug(f"Input format: {pix_fmt} with bit depth {bit_depth}.")
+
+        def _add_hwaccel_opts(_hwaccel: str, _decoder: str):
+            # set hwaccel and hwaccel_output_format, if specified
+            opts.extend(['-hwaccel', _hwaccel])
+            if self.settings.hwoutputfmt.get(_hwaccel):
+                opts.extend(['-hwaccel_output_format', self.settings.hwoutputfmt[_hwaccel]])
+
+            # set hw device, if specified
+            nonlocal device
+            device = self.settings.hwdevices.get(_hwaccel)
+            if device:
+                self.log.debug("Setting hwaccel device to %s." % device)
+                opts.extend(['-init_hw_device', '%s=sma:%s' % (_hwaccel, device)])
+                opts.extend(['-hwaccel_device', 'sma'])
+
+            # Set vcodec
+            opts.extend(['-vcodec', _decoder])
+
+        # If there's a manually specified hwaccel/decoder pairing for this codec, use it.
+        if video_codec in self.settings.hwaccel_decoder_override:
```

Comment:
For a specific input codec, users will be able to specify a `hwaccel` and a `decoder` using the format `<codec>:<hwaccel>.<decoder>`.

ex:
```
hwaccel_decoder_override = av1:vaapi.av1
```
Happy to add a bit to the wiki about this.